### PR TITLE
[backup] backup coordinator: epoch ending before other types

### PR DESCRIPTION
**Pull request by msmouse**
## Motivation
So that we will never hit the edge case where a proof in the other backup types fails to verify because required validator set is not yet backed up.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

existing coverage (smoke test) and manual verification on my laptop

### CODE REVIEW

File: storage/backup/backup-cli/src/backup_types/epoch_ending/restore.rs [hunk 1]
```diff
@@ -206,7 +206,7 @@ impl EpochHistory {
         ensure!(!self.epoch_endings.is_empty(), "Empty epoch history.",);
         ensure!(
             epoch <= self.epoch_endings.len() as u64,
-            "History until epoch {} can't varify epoch {}",
+            "History until epoch {} can't verify epoch {}",
```
Comment (lightmark) [inline storage/backup/backup-cli/src/backup_types/epoch_ending/restore.rs:209]: lmao

File: storage/backup/backup-cli/src/coordinators/backup.rs [hunk 2]
```diff
@@ -255,7 +258,7 @@ impl BackupCoordinator {
     ) -> impl StreamExt<Item = ()> + 'a
     where
         S: Copy + Debug + 'a,
-        W: Worker<'a, S, Fut> + Copy + 'static,
+        W: Worker<'a, S, Fut> + Copy + 'a,
```
Comment (lightmark) [inline storage/backup/backup-cli/src/coordinators/backup.rs:261]: no idea what you are doing here.

**Review by lightmark — APPROVED** [review]

File: storage/backup/backup-cli/src/coordinators/backup.rs (hunk 2 above)
Comment (msmouse) [reply storage/backup/backup-cli/src/coordinators/backup.rs:261]: Now there's a closure passed in, so not always `'static` any more..

**Pull request closed without merging**